tests/unit/swagger/definitionsGeneration/definitions.spec.ts

```diff
@@ -830,6 +836,10 @@ describe('Definition generation', () => {
                   defaultValue1: { type: 'string', default: 'Default Value 1', description: undefined, format: undefined, example: undefined },
                   enumKeys: { type: 'string', default: undefined, description: undefined, format: undefined, example: undefined, enum: ['OK', 'KO'], 'x-nullable': false },
                   id: { type: 'number', format: 'double', default: undefined, description: undefined, example: undefined },
+                  indexedType: { type: 'string', default: undefined, description: undefined, format: undefined, example: undefined },
+                  indexedTypeToAlias: { $ref: '#/definitions/IndexedInterface', description: undefined, format: undefined, example: undefined },
```

Comment:
This should be `$ref`-ing the alias.
I believe you can get an `aliasSymbol` from the type and run that through the TypeResolver in these cases.

Reply:
I've tried but it seems that `aliasSymbol` property on type (`this.current.typeChecker.getTypeOfSymbolAtLocation`) is undefined.

Reply:
I found a little bit of time to have a look:

My main issue is that `this.current.typeChecker.getTypeOfSymbolAtLocation(symbol, this.typeNode.objectType)` is bad, because, as we already noticed, it does not preserve aliases.
However, you can get the Declatations of the symbol through `symbol.getDeclarations()`. I assume this is a `PropertyDeclaration` for demonstration:

 ```diff
- return new TypeResolver(this.current.typeChecker.typeToTypeNode(declaration)!
+ return new TypeResolver((symbol.getDeclarations()![0] as ts.PropertyDeclaration).type!
```

will already handle all the cases you added to the tests.

Reply:
Great! Thank you it worked